internal/graph/shadow_resolver.go

```diff
@@ -67,20 +67,22 @@ func (s ShadowResolver) ResolveCheck(ctx context.Context, req *ResolveCheckReque
 		reqClone.VisitedPaths = nil // reset completely for evaluation
 		s.wg.Add(1)
 		go func() {
-			ctx, cancel := context.WithTimeout(ctxClone, s.shadowTimeout)
 			defer func() {
```

Comment:
A defer function with a recover should not include any additional code that may panic because it cannot capture it. It should also always be the first defer function, so that it is the last to run.